sync-gateway-config.json

```diff
@@ -125,10 +125,11 @@ function(doc, oldDoc){
     // Add doc to task-list and moderators channel.
     channel("task-list." + doc.taskList.id);
     channel("moderators");
-  } else if (doc.type == "task-list.user") {
+  } else if (doc.type == "task-list.user" || (oldDoc && oldDoc.type == "task-list.user")) {
     /* Control Write Access */
+    var owner = doc._deleted ? oldDoc.taskList.owner : doc.taskList.owner;
     try {
-      requireUser(doc.taskList.owner);
+      requireUser(owner);
     } catch (e) {
       requireRole("moderator");
```

Comment:
Yes it was initially abstracted away and we also had `requireUserOrAccess`. It makes it harder to explain what is going on when having inner functions like this. That's why i changed them to be inline and just focused on the `try/catch` pattern. I'll keep this in mind and try to figure something out.